### PR TITLE
Remove Reading and Writing structs and remove cloning of MsgEnvelope

### PR DESCRIPTION
**Pull request by lionel1704**

### CODE REVIEW

File: src/node/adult_duties/chunks/reading.rs [hunk 1]
```diff
@@ -9,55 +9,38 @@
...
             None
         }
-    }
-
-    fn verify_msg(&self) -> bool {
-        match self.msg.most_recent_sender() {
-            MsgSender::Section { .. } => self.msg.verify(),
-            _ => false,
-        }
+    // } else if matches!(self.requester, PublicId::Node(_)) {
```
Comment (dirvine) [inline src/node/adult_duties/chunks/reading.rs:29]: @lionel1704  did you mean to leave this commented block?

File: src/node/adult_duties/chunks/writing.rs [hunk 2]
```diff
@@ -6,53 +6,36 @@
...
         }
-    }
-
-    fn verify_msg(&self) -> bool {
-        match self.msg.most_recent_sender() {
-            MsgSender::Section { .. } => self.msg.verify(),
-            _ => false,
+        DeletePrivate(address) => {
+            if msg.verify() {
+                // really though, for a delete, what we should be looking at is the origin signature! That would be the source of truth!
```
Comment (dirvine) [inline src/node/adult_duties/chunks/writing.rs:33]: This is worth "fixing"

File: src/node/adult_duties/chunks/reading.rs (hunk 1 above)
Comment (lionel1704) [reply src/node/adult_duties/chunks/reading.rs:29]: Yes @dirvine 
This code is a part of the data duplication process when a node leaves the network. That part has been temporarily disabled in the refactor. So we'll need to enable this later.

File: src/node/adult_duties/chunks/writing.rs (hunk 2 above)
Comment (lionel1704) [reply src/node/adult_duties/chunks/writing.rs:33]: Yes. But the change was not a part of this PR. So I believe it should be in a PR of it's own with other relevant changes?

**Review by Yoga07 — APPROVED** [review]

**Pull request merged**